src/scripts/archive_dataset/archive_dataset.py

```diff
@@ -0,0 +1,190 @@
+a single dataset, thus archives have an update number appended
+to their name. For example, if this is the third time we are
+archiving the `dataset_metadata_v2` dataset, the archive will be named
+`dataset_metadata_v2_3`. The script will automatically determine which
+update number this is, therefore it is not necessary to pass the
+update number as a command line argument.
+
+Datasets are assumed to exist in:
+
+    s3://{bucket}/{app}/{parquet}/
+
+And archives will be stored to:
+
+    s3://{bucket}/{app}/{parquet}/archive/
+
+When relationalizing a dataset, as is done in the json_to_parquet jobs,
+descendent datasets (such as `dataset_metadata_files`, a descendent
+of `dataset_metadata`) are created. For ease of use with the typical
+use case, descendent datasets are archived when passing the root
+dataset to `--dataset`. For example, passing `--dataset=dataset_metadata`
+and `--dataset-version=v2` will archive both `dataset_metadata_v2` and
+`dataset_metadata_files_v2`.
+
+Archive updates are tracked at the root dataset level.
+For example, if a struct `folders` is added to the `metadata_v2`
+JSON dataset in that dataset's second compatible schema update, then
+the next time a compatible schema update occurs the third archive
+operation will also occur. Although this is the first time that the
+`metadata_folders_v2` dataset will be archived, its archive will be
+named `metadata_folders_v2_3`.
+"""
+import argparse
+import os
+import shlex
+import subprocess
+import boto3
+
+
+def read_args():
+    """Read command line arguments."""
+    parser = argparse.ArgumentParser(
+            description=("Archive a parquet dataset for each study of an app "
+                         "by copying it to the standard archive location: "
+                         "s3://{--bucket}/{--app}/{study}/parquet/archive/"
+                         "{--dataset}_{--dataset-version}_{update-number}/"))
+    parser.add_argument("--bucket",
+                        help="The name of the S3 bucket.")
+    parser.add_argument("--app",
+                        help="The app identifier.")
+    parser.add_argument("--dataset",
+                        help="The dataset name.")
+    parser.add_argument("--dataset-version",
+                        help="The dataset name.")
+    parser.add_argument("--profile",
+                        help="The AWS profile to use.")
+    args = parser.parse_args()
+    return args
+
+
+def get_source_and_dest_prefix(s3_client, bucket, app, dataset, dataset_version):
+    """Construct the source and destination S3 prefixes of the archives.
+
+    Keyword arguments:
+    s3_client -- An AWS S3 client object.
+    bucket -- The S3 bucket name.
+    app -- The app identifier, as used in the S3 prefix in the bucket, to archive.
+    dataset -- The name of the dataset to archive.
+    dataset_version -- The dataset version to archive.
+
+    Returns:
+    A dictionary with source prefixes as keys and destination
+    prefixes as values.
```

Comment:
Just out of curiosity, where did you find this python docstring standard?  I'm only familiar with
* google
* pydoc
* numpy

Reply:
https://www.python.org/dev/peps/pep-0257/